pkg/reconciler/v1alpha1/testing/functional.go

```diff
@@ -160,6 +160,16 @@ func WithVolume(name, mountPath string, volumeSource corev1.VolumeSource) Servic
 	}
 }
 
+// WithIngressClassAnnotation assigns network ingress class annotation to a service
```

Comment:
I think you should instead add a more generic
`WithServiceAnnotations` method that takes `map[string]string` and does a `resources.UnionMaps` call (will also deal with the empty map).

Reply:
done